packages/did/src/DidDetails/LightDidDetails.ts

```diff
@@ -132,10 +136,13 @@ function serializeAdditionalLightDidDetails({
     objectToSerialize[KEY_AGREEMENT_MAP_KEY] = key
   }
   if (service && service.length > 0) {
-    objectToSerialize[SERVICES_MAP_KEY] = service.map(({ id, ...rest }) => ({
-      id: resourceIdToChain(id),
-      ...rest,
-    }))
+    objectToSerialize[SERVICES_MAP_KEY] = service.map(
+      ({ id, serviceEndpoint, type }) => ({
+        id: resourceIdToChain(id),
+        serviceEndpoint,
+        type,
+      })
+    )
```

Comment:
hmm how does this help? I think it was more compatible the way it was before. This way we'll just end up with serialised service endpoints where serviceEndpoint is undefined and url is silently dropped.

Reply:
This is serialization of the new stuff. So new endpoints always serialize to id, serviceEndpoint, type. The trick is in the DEserialization, which attemps to use serviceEndpoint and type, and falls back to types and urls if those are not found.

Reply:
In this snippet here I replaced `...rest`, which was anyway just `serviceEndpoint` and `type` with its building components.

Reply:
Well not if you're dealing with a legacy DID, then it's the legacy keys

Reply:
although the services validation runs beforehand and should not allow the use of any other keys 🤔 anyway, it's good that we reverted it bc that was not the issue and the remainder syntax is less brittle.